client/web/compose/src/components/Admin/Module/UniqueValues.vue

```diff
@@ -0,0 +1,212 @@
+              :get-option-label="getOptionLabel"
+              :options="filterFieldOptions(rule)"
+              class="bg-white"
+            />
+
+            <b-input-group-append>
+              <b-button
+                variant="primary"
+                class="px-4"
+                @click="updateRuleConstraint(rule)"
+              >
+                {{ $t("add") }}
+              </b-button>
+            </b-input-group-append>
+          </b-input-group>
+        </b-form-group>
+      </div>
+    </div>
+
+    <hr>
+
+    <div class="d-flex justify-content-end">
+      <b-button
+        size="lg"
+        variant="link"
+        class="d-flex align-items-center text-decoration-none p-0 mt-3"
+        @click="addNewConstraint"
+      >
+        <font-awesome-icon
+          :icon="['fas', 'plus']"
+          class="mr-2"
+        />
+        {{ $t("addNewConstraint") }}
+      </b-button>
+    </div>
+  </div>
+</template>
+
+<script>
+import { compose } from '@cortezaproject/corteza-js'
+import { VueSelect } from 'vue-select'
+
+export default {
+  i18nOptions: {
+    namespaces: 'module',
+    keyPrefix: 'edit.config.uniqueValues',
+  },
+
+  components: {
+    VueSelect,
+  },
+
+  props: {
+    module: {
+      type: compose.Module,
+      required: true,
+    },
+  },
+
+  computed: {
+    rules: {
+      get () {
+        return this.module.config.recordDeDup.rules
+      },
+      set (value) {
+        this.module.config.recordDeDup.rules = value
+      },
+    },
+
+    modifierOptions () {
+      return [
+        { value: 'ignore-case', text: this.$t('ignoreCase') },
+        { value: 'fuzzy-match', text: this.$t('fuzzyMatch') },
+        { value: 'sounds-like', text: this.$t('soundsLike') },
+        { value: 'case-sensitive', text: this.$t('caseSensitive') },
+      ]
+    },
+
+    multiValueOptions () {
+      return [
+        { value: 'one-of', text: this.$t('oneOf') },
+        { value: 'equal', text: this.$t('equal') },
+      ]
+    },
+  },
+
+  methods: {
+    addNewConstraint () {
+      this.rules.push({
+        name: '',
+        strict: true,
+        constraints: [],
+      })
+    },
+
+    updateRuleConstraint (rule) {
+      rule.constraints.push({
+        attribute: rule.currentField.name,
+        modifier: 'case-sensitive',
+        multiValue: 'equal',
+        type: rule.currentField.kind,
+        isMulti: rule.currentField.isMulti,
+      })
+
+      rule.currentField = undefined
+    },
+
+    filterFieldOptions (rule) {
+      const selectedFields = rule.constraints.map(({ attribute }) => attribute)
+      return this.module.fields.filter(({ name }) => !selectedFields.includes(name))
+    },
+
+    getField (attribute) {
+      const field = this.module.fields.find(
+        ({ name }) => name === attribute,
+      )
+
+      return field || {}
+    },
+
+    getFieldLabel (constraint) {
+      return this.getField(constraint.attribute).label || this.getField(constraint.attribute).name
```

Comment:
You forgot to use getOptionLabel here, or even better just use getOptionLabel(getField()) in the template